static/app/components/contentSliderDiff/index.tsx

```diff
@@ -0,0 +1,269 @@
+
+interface DiffSidesProps
+  extends Pick<
+    ContentSliderDiffProps,
+    'onDividerMouseDown' | 'beforeContent' | 'afterContent'
+  > {
+  viewDimensions: {height: number; width: number};
+  width: string | undefined;
+}
+
+function DiffSides({
+  onDividerMouseDown,
+  viewDimensions,
+  width,
+  beforeContent,
+  afterContent,
+}: DiffSidesProps) {
+  const beforeElemRef = useRef<HTMLDivElement>(null);
+  const dividerElem = useRef<HTMLDivElement>(null);
+
+  const resizableDrawer = useResizableDrawer({
+    direction: 'left',
+    initialSize: viewDimensions.width / 2,
+    min: 0,
+    onResize: newSize => {
+      const maxWidth = viewDimensions.width - BORDER_WIDTH;
+      if (beforeElemRef.current) {
+        beforeElemRef.current.style.width =
+          viewDimensions.width === 0
+            ? '100%'
+            : (toPixels(Math.max(BORDER_WIDTH, Math.min(maxWidth, newSize))) ?? '0px');
+      }
+      if (dividerElem.current) {
+        dividerElem.current.style.left =
+          toPixels(Math.max(BORDER_WIDTH, Math.min(maxWidth, newSize))) ?? '0px';
+      }
+    },
+  });
+
+  const handleDividerMouseDown: React.MouseEventHandler<HTMLElement> = useCallback(
+    (event: React.MouseEvent<HTMLElement>) => {
+      onDividerMouseDown?.(event);
+      resizableDrawer.onMouseDown(event);
+    },
+    [onDividerMouseDown, resizableDrawer]
+  );
+
+  return (
+    <Fragment>
+      <Cover style={{width}} data-test-id="after-content">
+        <Placement style={{width}}>
+          <FullHeightContainer>{afterContent}</FullHeightContainer>
+        </Placement>
+      </Cover>
+      <Cover ref={beforeElemRef} data-test-id="before-content">
+        <Placement style={{width}}>
+          <FullHeightContainer>{beforeContent}</FullHeightContainer>
+        </Placement>
+      </Cover>
+      <Divider
+        data-test-id="divider"
+        ref={dividerElem}
+        onMouseDown={handleDividerMouseDown}
+      />
+    </Fragment>
+  );
+}
+
+const DiffHeader = styled('div')`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: ${space(1)};
+  font-weight: ${p => p.theme.fontWeightBold};
+  line-height: 1.2;
+  justify-content: space-between;
+
+  & > *:first-child {
+    color: ${p => p.theme.red300};
+  }
+
+  & > *:last-child {
+    color: ${p => p.theme.green300};
+  }
+  margin-bottom: ${space(0.5)};
+`;
+
+const Label = styled('div')`
+  display: flex;
+  gap: ${space(0.5)};
+  align-items: center;
+  font-weight: bold;
+`;
+
+const FullHeightContainer = styled(NegativeSpaceContainer)`
+  height: 100%;
+`;
+
+const OverflowVisibleContainer = styled(FullHeightContainer)`
+  overflow: visible;
+`;
+
+const Positioned = styled('div')`
+  height: 100%;
+  position: relative;
+  width: 100%;
+`;
+
+const Divider = styled('div')`
+  --handle-size: ${space(1.5)};
+  --line-width: 1px;
+
+  cursor: ew-resize;
+  width: var(--line-width);
+  height: 100%;
+  background: ${p => p.theme.purple400};
```

Comment:
we probably shouldn’t be using the `theme` colors directly, as they might not work with the chonk re-design. It would be better if we could use semantic tokens instead of the `purple` palette here.